libs/gltfio/src/FNodeManager.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GLTFIO_FNODEMANAGER_H
+#define GLTFIO_FNODEMANAGER_H
+
+#include "upcast.h"
+
+#include <gltfio/NodeManager.h>
+
+#include <utils/compiler.h>
+#include <utils/SingleInstanceComponentManager.h>
+#include <utils/Entity.h>
+#include <utils/Log.h>
+#include <utils/Slice.h>
+
+namespace gltfio {
+
+class UTILS_PRIVATE FNodeManager : public NodeManager {
+public:
+    using Instance = NodeManager::Instance;
+
+    FNodeManager() noexcept {}
+
+    ~FNodeManager() noexcept {
+        assert_invariant(mManager.getComponentCount() == 0);
+    }
+
+    void terminate() noexcept {
+        auto& manager = mManager;
+        if (!manager.empty()) {
+    #ifndef NDEBUG
+            utils::slog.d << "cleaning up " << manager.getComponentCount()
+                << " leaked node components" << utils::io::endl;
+    #endif
+            while (!manager.empty()) {
+                Instance ci = manager.end() - 1;
+                manager.removeComponent(manager.getEntity(ci));
+            }
+        }
+    }
```

Comment:
don't put the implementation in the header file.

Reply:
Done.